File: branches/Medusa/Source/Fantasy.Jobs/Resources/TaskCountSettings.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml.Serialization;
using Fantasy.Configuration;

namespace Fantasy.Jobs.Resources
{
   
    public class TaskCountSettings 
    {
        public TaskCountSettings()
        {
            Count = Int32.MaxValue;
        }

        [XmlAttribute("count")]
        public int Count { get; set; }


        private List<TaskCountSetting> _tasks = new List<TaskCountSetting>();

        [XmlArray("tasks"),
        XmlArrayItem(Type=typeof(TaskCountSetting), ElementName="task") ] 
        public List<TaskCountSetting> Tasks
        {
            get { return _tasks; }
        }

        public override bool Equals(object obj)
        {
            return ComparsionHelper.DeepEquals(this, obj);
        }

        public override int GetHashCode()
        {
            return base.GetHashCode();
        }
    }
}

File: Fantasy.Jobs/src/fantasy/jobs/resources/TaskCountSettings.java
package fantasy.jobs.resources;

import Fantasy.Configuration.*;

public class TaskCountSettings
{
	public TaskCountSettings()
	{
		setCount(Integer.MAX_VALUE);
	}

//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
	//[XmlAttribute("count")]
	private int privateCount;
	public final int getCount()
	{
		return privateCount;
	}
	public final void setCount(int value)
	{
		privateCount = value;
	}


	private java.util.ArrayList<TaskCountSetting> _tasks = new java.util.ArrayList<TaskCountSetting>();

//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
	//[XmlArray("tasks"), XmlArrayItem(java.lang.Class=typeof(TaskCountSetting), ElementName="task")]
	public final java.util.ArrayList<TaskCountSetting> getTasks()
	{
		return _tasks;
	}

	@Override
	public boolean equals(Object obj)
	{
		return ComparsionHelper.DeepEquals(this, obj);
	}

	@Override
	public int hashCode()
	{
		return super.hashCode();
	}
}